pkg/cmd/controller/pipelinerunner.go

```diff
@@ -457,3 +464,42 @@ func (o *PipelineRunnerOptions) setupSignalChannel(cancel context.CancelFunc) {
 		cancel()
 	}()
 }
+
+func (o *PipelineRunnerOptions) getSourceURL(org, repo string) string {
+	jxClient, ns, err := o.getClientsAndNamespace()
+
+	resourceInterface := jxClient.JenkinsV1().SourceRepositories(ns)
```

Comment:
given that all you need _jxClient_ in this function is for getting the source repository interface, you might as well push that into `getClientsAndNamespace` and call it `getSourceRepositoryInterface`. Up to you really.

Reply:
`getClientsAndNamespace` is much general that may be used for some other case later.  So I wanna keep it.